### PR TITLE
[Concept exercise] Chessboard - ranges

### PR DESCRIPTION
**Pull request by neenjaw**
Port from Elixir/chessboard

### CODE REVIEW

File: exercises/concept/chessboard/chessboard_test.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+require 'minitest/autorun'
+require_relative 'chessboard.rb'
+
+class ChessboardTest < Minitest::Test
+  def test_rank_range_is_a_range_of_integers
+    assert_equal 1..8, Chessboard.rank_range
+  end
+
+  def test_file_range_is_a_range_of_letters
+    assert_equal ?A..?H, Chessboard.file_range
```
Comment (kotp) [inline exercises/concept/chessboard/chessboard_test.rb:12]:
```suggestion
    assert_equal ('A'..'H'), Chessboard.file_range
```

Same thing, but probably more familiar to a student.  If we have introduced this earlier in some concept about range or strings, then this suggestion is not necessary to consider.

File: exercises/concept/chessboard/.docs/instructions.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# Instructions
+
+As a chess enthusiast, you would like to write your own version of the game. Yes, there maybe plenty of implementations of chess available online already, but yours will be unique!
+
+But before you can let your imagination run wild, you need to take care of the basics. Let's start by generating the board.
+
+Each square of the chessboard is identified by a letter-number pair. The vertical columns of squares, called files, are labeled A through H. The horizontal rows of squares, called ranks, are numbered 1 to 8.
```
Comment (kotp) [inline exercises/concept/chessboard/.docs/instructions.md:7]:
```suggestion
Each square of the chessboard is identified by a letter-number pair. The vertical columns of squares, called files, are labeled `"A"` through `"H"`. The horizontal rows of squares, called ranks, are numbered `1` to `8`.
```

File: exercises/concept/chessboard/.docs/instructions.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+## 1. Define the rank range
+
+Implement the `rank_range` method. It should return a range of integers, from 1 to 8.
```
Comment (kotp) [inline exercises/concept/chessboard/.docs/instructions.md:11]:
```suggestion
Implement the `rank_range` method. It should return a range of integers, from `1` to `8`.
```

File: exercises/concept/chessboard/.docs/instructions.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+```ruby
+Chessboard.rank_range()
+```
+
+## 2. Define the file range
+
+Implement the `file_range` method. It should return a range of strings, from the uppercase letter A, to the uppercase letter H.
```
Comment (kotp) [inline exercises/concept/chessboard/.docs/instructions.md:19]:
```suggestion
Implement the `file_range` method. It should return a range of strings, from the uppercase letter `"A"`, to the uppercase letter `"H"`.
```

File: exercises/concept/chessboard/.docs/instructions.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+```ruby
+Chessboard.file_range()
+```
+
+## 3. Transform the rank range into a list of ranks
+
+Implement the `ranks` method. It should return a list of integers, from 1 to 8. Do not write the list by hand, generate it from the range returned by the `rank_range` method.
```
Comment (kotp) [inline exercises/concept/chessboard/.docs/instructions.md:27]:
```suggestion
Implement the `ranks` method. It should return a list of integers, from `1` to `8`. Do not write the list by hand, generate it from the range returned by the `rank_range` method.
```

File: exercises/concept/chessboard/.docs/instructions.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+```ruby
+Chessboard.ranks()
+# => [1, 2, 3, 4, 5, 6, 7, 8]
+```
+
+## 4. Transform the file range into a list of files
+
+Implement the `files` method. It should return a list of letters (strings), from "A" to "H". Do not write the list by hand, generate it from the range returned by the `file_range` method.
```
Comment (kotp) [inline exercises/concept/chessboard/.docs/instructions.md:36]:
```suggestion
Implement the `files` method. It should return a list of letters (strings), from `"A"` to `"H"`. Do not write the list by hand, generate it from the range returned by the `file_range` method.
```

File: exercises/concept/chessboard/chessboard_test.rb (hunk 1 above)
Comment (neenjaw) [reply exercises/concept/chessboard/chessboard_test.rb:12]: I think this change is a good one!

**Comment by kytrinyx** [comment]
It's been over 18 months since the last movement on this PR.

Does someone who was involved at the time have an idea of what it would take to get this over the finish line?

File: exercises/concept/chessboard/.docs/instructions.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
+# Instructions
+
+As a chess enthusiast, you would like to write your own version of the game. Yes, there maybe plenty of implementations of chess available online already, but yours will be unique!
```
Comment (kotp) [inline exercises/concept/chessboard/.docs/instructions.md:3]: Single Sentence Per Line as per [markdown specification] for Exercism.

[markdown specification]: https://exercism.org/docs/building/markdown/markdown

```suggestion
As a chess enthusiast, you would like to write your own version of the game.
Yes, there maybe plenty of implementations of chess available online already, but yours will be unique!
```

**Review by kotp — COMMENTED** [review]
Still in draft, so a comment, rather than request for change.

**Pull request closed without merging**